data/src/main/java/org/apache/iceberg/data/avro/IcebergDecoder.java

```diff
@@ -190,6 +190,7 @@ public D decode(InputStream stream, D reuse) {
    * @return true if the buffer is complete, false otherwise (stream ended)
    * @throws IOException if there is an error while reading
    */
+  @SuppressWarnings("checkstyle:InnerAssignment")
```

Comment:
Curious, is there a way we can get around this without suppressing?

Reply:
I tried it, but there wasn't a good way without changing the familiar idiom of stream reading.

Reply:
This suppression looks okay to me.